js/derivadex.js

```diff
@@ -0,0 +1,913 @@
+        const market = this.market (symbol);
+        const tickers = await this.fetchTickers ([ market['symbol'] ], params);
+        const ticker = this.safeValue (tickers, market['symbol']);
+        if (ticker === undefined) {
+            throw new BadSymbol (this.id + ' fetchTicker() symbol ' + symbol + ' not found');
+        }
+        return ticker;
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        /**
+         * @method
+         * @name derivadex#fetchTickers
+         * @description fetches price tickers for multiple markets, statistical calculations with the information calculated over the past 24 hours each market
+         * @param {[string]|undefined} symbols unified symbols of the markets to fetch the ticker for, all market tickers are returned if not assigned
+         * @param {object} params extra parameters specific to the derivadex api endpoint
+         * @returns {object} an array of [ticker structures]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        symbols = this.marketSymbols (symbols);
+        const result = {};
+        for (let i = 0; i < symbols.length; i++) {
+            const ticker = await this.constructTicker (symbols[i]);
+            if (ticker !== undefined) {
+                result[symbols[i]] = ticker;
+            }
+        }
+        return result;
+    }
+
+    async constructTicker (symbol) {
+        const params = {};
+        params['symbol'] = symbol;
+        params['depth'] = 1;
+        const request = {
+            'symbol': symbol,
+        };
+        const orderBookResponse = await this.publicGetMarketsOrderBookL2Symbol (this.extend (request, params)); // markets/order_book endpoint response is cached for 10 seconds
+        const orderBookValue = orderBookResponse['value'];
+        const bid = this.safeString (orderBookValue[0], 'price');
+        const bidVolume = this.safeString (orderBookValue[0], 'amount');
+        const ask = this.safeString (orderBookValue[1], 'price');
+        const askVolume = this.safeString (orderBookValue[1], 'amount');
+        const volumeParams = {};
+        volumeParams['symbol'] = symbol;
+        volumeParams['aggregationPeriod'] = 'day';
+        const volumeAggregationResponse = await this.publicGetAggregationsVolume (volumeParams);
+        const volumeValue = volumeAggregationResponse['value'][0];
+        const volumeKey = 'volume_' + symbol;
+        const tickerResponse = await this.publicGetMarketsTickers ({ 'symbol': symbol });
+        const ticker = tickerResponse['value'][0];
+        const timestamp = this.safeString (tickerResponse, 'timestamp');
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeString (ticker, 'high'),
+            'low': this.safeString (ticker, 'low'),
+            'bid': bid,
```

Comment:
just checking that what the tickers endpoint is requesting is the most recent bid and ask (and bid and ask volumes)? nothing to do with a lookback period right? if so, then what you're doing with the orderbookL2 makes sense